File: 560/560_version1.py
class Solution(object):
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        dic = {}
        dic[0] = 1
        csum,cnt = 0,0
        for n in nums:
            csum+=n
            cnt+=dic.get(csum-k,0)
            dic[csum] = dic.get(csum,0)+1
        return cnt

File: 560/560_version1.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        if(nums.length==0) return 0;
        int cNum = 0,cSum = 0;
        HashMap<Integer,Integer> cache = new HashMap<>();
        cache.put(0,1);
        for(int n:nums){
            cSum+=n;
            if(cache.containsKey(cSum-k)) cNum+=cache.get(cSum-k);
            if(cache.containsKey(cSum)) cache.put(cSum,cache.get(cSum)+1);
            else cache.put(cSum,1);
        }
        return cNum;
    }
}
